plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2933,6 +2937,39 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
         commitUpdateAndTransaction(rowDelta, session, transaction, "write");
     }
 
+    private TupleDomain<IcebergColumnHandle> extractTupleDomainsFromCommitTasks(IcebergTableHandle table, Table icebergTable, List<CommitTaskData> commitTasks)
+    {
+        List<IcebergColumnHandle> partitionColumns = getProjectedColumns(icebergTable.schema(), typeManager, identityPartitionColumnsInAllSpecs(icebergTable));
```

Comment:
Do we have a test that verifies partitioning by nested fields?

Reply:
`testConcurrentNonOverlappingUpdateOnNestedPartition` should cover that